upload_log_files.py

```diff
@@ -0,0 +1,121 @@
+        uploaded_log_dates.append(date_match.group())
+
+    return uploaded_log_dates
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Uploads pipeline log files to g-cloud")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file-path",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("memory_profile_dir_path", metavar="memory-profile-dir-path",
+                        help="Path to the memory profile log directory with file to upload")
+    parser.add_argument("data_archive_dir_path", metavar="data-archive-dir-path",
+                        help="Path to the data archive directory with file to upload")
+
+    args = parser.parse_args()
+
+    user = args.user
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    memory_profile_dir_path = args.memory_profile_dir_path
+    data_archive_dir_path = args.data_archive_dir_path
+
+    date_pattern = r'\d{4}-\d{2}-\d{2}'
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    log.warning(f"Deleting old memory profile & data archives files from local disk")
+    delete_old_log_files(memory_profile_dir_path, fetch_latest_modified_file_path(memory_profile_dir_path))
+    delete_old_log_files(data_archive_dir_path, fetch_latest_modified_file_path(data_archive_dir_path))
+
+    uploaded_memory_logs = list_blobs(pipeline_configuration.memory_profile_upload_bucket,
+                                      pipeline_configuration.log_dir_path, google_cloud_credentials_file_path)
+    uploaded_memory_log_dates = get_uploaded_log_dates(uploaded_memory_logs, date_pattern)
+    uploaded_data_archives = list_blobs(pipeline_configuration.memory_profile_upload_bucket,
+                                        pipeline_configuration.log_dir_path, google_cloud_credentials_file_path)
+    uploaded_data_archives_dates = get_uploaded_log_dates(uploaded_data_archives, date_pattern)
+
+    latest_memory_log_file_path = fetch_latest_modified_file_path(memory_profile_dir_path)
+    memory_profile_upload_location = f"{pipeline_configuration.memory_profile_upload_bucket}/" \
+        f"{pipeline_configuration.log_dir_path}{os.path.basename(latest_memory_log_file_path)}"
```

Comment:
```suggestion
        f"{pipeline_configuration.log_dir_path}/{os.path.basename(latest_memory_log_file_path)}"
```
I think safer to include a path separator here